src/System.Management.Automation/engine/parser/TypeResolver.cs

```diff
@@ -302,10 +321,28 @@ internal static Type ResolveTypeNameWithContext(TypeName typeName, out Exception
                 return typeName._typeDefinitionAst.Type;
             }
 
-            result = ResolveTypeNameWorker(typeName, currentScope, typeResolutionState.assemblies, typeResolutionState, /* reportAmbiguousException */ true, out exception);
+            var assemList = assemblies ?? ClrFacade.GetAssemblies(typeResolutionState, typeName);
+            // A set of assembly names that we have searched from but found no matching type.
+            // This set will be updated within ResolveTypeNameWorker.
+            var searchedAssemblies = new HashSet<string>();
```

Comment:
I'm not sure it should be case insensitive, and somewhat worse, I'm not sure the FullName can be considered unique.

As an example, create a PowerShell class at the command line multiple times.  You'll see multiple assemblies with the same assembly name. And suggesting we change this in PowerShell doesn't help - anyone could create different dynamic assemblies with identical names.